src/sysen3/papsav.1

```diff
@@ -0,0 +1,372 @@
+        addi b,60
+        tyo b
+        popj p,
+
+typhaf: jsp t,.entr.
+        0
+        move a,@(.ap.)
+        tlnn a,-1		; if has left half
+        jrst typhaf+12
+        hlrz b,a
+        movei t1,nogood+35
+        pushj p,typoct
+        tyo [",]
+        tyo [",]
+        hrrzs a
+        movei t1,nogood+32
+        pushj p,typoct
+        jrst .retn.
+
+tycrlf: jsp t,.entr.
+        0
+        movei t1,[crlf]
+        pushj p,type
+        jrst .retn.
+
+typsix: jsp t,.entr.
+        0
+        move b,@(.ap.)
+6loop:  setz a,
+        lshc a,6
+        addi a,40
+        tyo a
+        jumpn b,6loop
+        jrst .retn.
+
+typasz: jsp t,.entr.
+        0
+        move a,@(.ap.)		; can't indirect param
+        movei t1,[36,,-11]
+        pushj p,type
+        jrst .retn.
+
+;;; PURIFICATION RITUAL
+
+pure:   0
+oitsvr: 0			;ITS version purified for
+ousrst: 0			;extra check
+
+ritual: jsp t,.entr.
+        0
+        skipe pure
+        jrst ritual+5
+        pushj p,purify
+	MOVE TT, [SQUOZE 0,ITSVRS]	;SEE IF SYSTEM CHANGED
+        .eval tt,
+        jfcl
+	MOVE T, [SQUOZE 0,USRSTG]	;see if user page map changed
+        .eval t,
+        jfcl
+        camn tt,oitsvr
+        came t,ousrst
+        jrst nogood
+        jrst .retn.
+
+nogood: movei t1,[bletch]	; wrong system version, purify
+        pushj p,type
+        jrst kill
+
+	constants
+
+        loc 602
+
+abstb1: 
+	<squoze 0,SYSMBF>	; system message buffer
+        sysmsg+7				
+        <squoze 0,TOIP>		; tty output ptr
+        syget0			
+        <squoze 0,TOBEP>	; end of buffer
+        syget0+2
+        <squoze 0,TOOP>		; output buffer output pointer
+        papsav+1
+
+immeds: <squoze 0,TOBL>		; tty output buffer length
+        syget0+3
+        <squoze 0,SYSCON>	; system tty number
+        papsav
+        <squoze 0,SYSMLN>	; log 2 of number of 4-word blocks
+        sysmsg+6
+
+purify: jsp t,.entr.
+        0
+        movei f,pt
+        movsi .prsva,-7
+evloop: move v,abstb1(i)		; get symbol
+        .eval v,			; evaluate it
+        .value				; not there???
+        caig lim,(i)			; skip following code
+        jrst evlp00			;  if immediate symbol
+        caige v,remapt			; remap low core
+        subi v,remapt			; into high core
+evlp00: addi i,t1
+        move tt,abstb1(i)		; follow patch list
+ploop:  skipn l,tt
+        jrst evlpnx
+        hrrz tt,(l)			; loc to patch
+        hrrm v,(l)			; patch it
+        jrst ploop			; and try again
+evlpnx: aobjn i,evloop			; next symbol
+
+        setom pure
+	MOVE v, [SQUOZE 0,ITSVRS]	; get its version symbol
+        .eval v,			; evaluate it
+        .value				; what???
+        movem v,oitsvr			; store its version
+	MOVE v, [SQUOZE 0,USRSTG]	;see if user page map changed
+        .eval v,			
+        .value
+        movem v,ousrst
+        setzm v
+        movei tt,t1
+        hrli tt,-177
+        move t,tt
+getmor: 
+	.call [setz ? sixbit/CORBLK/ ? 1000,,210000 ? 1000,,-1 ? tt ? 1000,,400000 ? setz t]
```

Comment:
- First argument should be %climm,,+%cbred%cbndr
- Second: %climm,,%jself
- Third: %climm,,400000

Reply:
Ok.

Reply:
+%cbred%cbndr doesn't appear to equate to 210000.  It results in 200000.  The resulting executable doesn't compare with RAKASH PAPSAV.

Reply:
I think cbndr should be %CBNDW

Reply:
Is the "+" misplaced?  Should it be %cbred+%cbndr

Reply:
Yes, the + should be between the two bit values.

Reply:
Yes, the initial + was my mistake.